udm-le.sh

```diff
@@ -76,10 +75,17 @@ initial)
 	fi
 
 	echo 'Attempting initial certificate generation'
-	${PODMAN_CMD} ${LEGO_ARGS} --accept-tos run && deploy_cert
+	${PODMAN_CMD} ${LEGO_ARGS} --accept-tos run && deploy_cert && add_captive && unifi-os restart
 	;;
 renew)
 	echo 'Attempting certificate renewal'
 	${PODMAN_CMD} ${LEGO_ARGS} renew --days 60 && deploy_cert
+	if [ "${NEW_CERT}" = "yes" ]; then
+		add_captive && unifi-os restart
+	fi
+	;;
+bootrenew)
+	echo 'Attempting certificate renewal'
+	${PODMAN_CMD} ${LEGO_ARGS} renew --days 60 && deploy_cert && add_captive && unifi-os restart
```

Comment:
Maybe I'm trying to think before I've had enough coffee, but I'm having trouble following this:

* When bootrenew gets called on boot, if PODMAN_CMD returns false because there's no new cert isn't it going to just stop and never run the rest of this? 
* If you call `renew` under normal circumstances, isn't it going to run ad_captive and restart the container every night?

Reply:
So my logic is the following: 

When bootrenew) gets called on boot, the PODMAN_CMD will run and check for the need to update the certificate. Even if the certificate doesn't need to be updated, the return condition (I believe) isn't false and so the deploy_cert function will run, see that there is no new cert, return out and then the add_captive function will run, and finally will restart unifi-os (which may be the only time its technically unneeded).   If the cert is updated, all the appropriate steps will occur as well. 

I could only test running the boot script with a certificate that didn't require updating, and confirmed that the add_captive and unifi-os restart occurred properly. 

When the cronjob calls renew), and the certificate is renewed, a variable called NEW_CERT is set to yes, which will invoke the add_captive and unifi-os restart... otherwise if there is no new certificate that night, nothing is changed. 

Does that make sense or does something still seem off?

Reply:
Ok, yeah that makes sense. I had forgotten that I made deploy_cert a no-op unless the find returned true and wasn't relying on the return value of the podman command. I think this looks good and unless UI changes the way the captive portal certs get regenerated is probably the most elegant way to deal given all the limitations. Thanks for sticking with this and seeing it through!